src/jit/morph.cpp

```diff
@@ -13472,13 +13501,25 @@ GenTree* Compiler::fgMorphSmpOp(GenTree* tree, MorphAddrContext* mac)
                         temp->AsLclFld()->gtFieldSeq =
                             GetFieldSeqStore()->Append(temp->AsLclFld()->gtFieldSeq, fieldSeq);
                     }
-                    else
+                    else // we have a GT_LCL_VAR
                     {
-                        temp->ChangeOper(GT_LCL_FLD); // Note that this makes the gtFieldSeq "NotAField"...
+                        assert(temp->OperGet() == GT_LCL_VAR);
+                        temp->ChangeOper(GT_LCL_FLD); // Note that this typically makes the gtFieldSeq "NotAField"...
```

Comment:
In what case would this *not* be `NotAField` - is this the zero-offset case?

Reply:
Yes, the code that I added with this change will set this:  (see the `compiler`.hpp diff)

```
                // Set the zeroFieldSeq in the GT_LCL_FLD node
                gtLclFld.gtFieldSeq = zeroFieldSeq;
```

Reply:
Might be worth adding that, e.g.
```
// Note that this typically makes the gtFieldSeq "NotAField", unless we have a zero-offset FieldSeq
```
or something like that.

Reply:
I still think it would be good to explain why it sometimes doesn't make it `NotAField`

Reply:
Sure I will add a comment explaining that case